### PR TITLE
Update flash info

### PR DESCRIPTION
**Pull request by kelvin-cao**
Update the flash_part_info_gen4 and data structure per revB fw change.

### CODE REVIEW

File: linux/switchtec.h [hunk 1]
```diff
@@ -138,17 +138,21 @@ enum {
...
+	SWITCHTEC_GEN4_MAP0_RUNNING = 0x00,
+	SWITCHTEC_GEN4_MAP1_RUNNING = 0x01,
+	SWITCHTEC_GEN4_KEY0_RUNNING = 0x02,
+	SWITCHTEC_GEN4_KEY1_RUNNING = 0x03,
+	SWITCHTEC_GEN4_BL2_0_RUNNING = 0x04,
+	SWITCHTEC_GEN4_BL2_1_RUNNING = 0x05,
+	SWITCHTEC_GEN4_CFG0_RUNNING = 0x06,
+	SWITCHTEC_GEN4_CFG1_RUNNING = 0x07,
+	SWITCHTEC_GEN4_IMG0_RUNNING = 0x08,
+	SWITCHTEC_GEN4_IMG1_RUNNING = 0x09,
```
Comment (lsgunth) [inline linux/switchtec.h]: Ugh, who lets the firmware people make unnecessary incompatible changes? That's a terrible practice. If they had simply added the key as 0x9 and 0xA this wouldn't be necessary and there wouldn't exist firmware and software that are now incompatible. Absolutely gross.

File: linux/switchtec_ioctl.h [hunk 2]
```diff
@@ -34,9 +34,13 @@
 #define SWITCHTEC_IOCTL_PART_VENDOR7	12
 #define SWITCHTEC_IOCTL_PART_BL2_0	13
 #define SWITCHTEC_IOCTL_PART_BL2_1	14
+#define SWITCHTEC_IOCTL_PART_MAP_0	15
+#define SWITCHTEC_IOCTL_PART_MAP_1	16
+#define SWITCHTEC_IOCTL_PART_KEY_0	17
+#define SWITCHTEC_IOCTL_PART_KEY_1	18
```
Comment (lsgunth) [inline linux/switchtec_ioctl.h]: These should probably be added in the patch that adds the code in the ioctl.

File: linux/switchtec.h (hunk 1 above)
Comment (kelvin-cao) [reply linux/switchtec.h]: It happens before RTP, though we can definitly avoid it. I'd expect they will keep it compatible afterward.

File: linux/switchtec_ioctl.h (hunk 2 above)
Comment (kelvin-cao) [reply linux/switchtec_ioctl.h]: updated

**Pull request merged**